Rebase to ocp/release:4.2.0-0.nightly-2019-07-12-041904

hardys opened the pull request:
Rebase on the latest openshift/installer now that some of our PRs have merged, and also to get an updated version of RHCOS that should unblock #100

dcbw committed:
aws: allow GENEVE (6081) and OVN database ports (6641 & 6642)

1) Allow GENEVE (6081) between all nodes (masters & workers)
2) Allow OVN databases (6641 & 6642) between all masters
3) Allow OVN databases (6641 & 6642) between masters and
   workers, but not between workers themselves

kikisdeliveryservice committed:
bootkube.sh: remove image refs for mco components

Remove references to mcc/mcd/mcs and setupetcd images now that the MCO has 1
super-image containing all of the sub-images

Requires: openshift/machine-config-operator#850
Related-to: openshift/machine-config-operator#739

patrickdillon committed:
pkg: Create cloud credentials secret for GCP.

Add GCP service account JSON to cloud cred secret.
For GCP the `cloud-cred-secret` is created with name `gcp-credentials` and stores the service account key in JSON format in the key `service_account.json`

patrickdillon committed:
Vendor golang client for GCP APIs.

Add vendor libraries for working with Google Cloud APIs. Updates dep Gopkg.toml for dependency on google.golang.org/api.

patrickdillon committed:
pkg: stub out GCP platform.

Adds GCP to platform switch cases so that gcp will be recognized as a valid platform. Logic in these switch cases still needs to be implemented.

patrickdillon committed:
asset/manifests: Create DNS manifest for GCP.

Add logic to write cluster DNS manifest for GCP. Instantiates a client to the GCP API, which retrieves a zone that matches the provided base domain.

* asset/installconfig: Adds GetPublicZone returns the public zone matching the basedomain
  if more than one public DNS Zones are found, the last such DNS Zone from the list provided by GCP APIs will be used.
  GCP list requires the DNS Name in the query to be a full qualifoed DNS name with `.` at the end

abhinavdahiya committed:
gcp: add GetSession

It reads the service account JSON content or file from following locations in order:
// env GOOGLE_CREDENTIALS,
// env GOOGLE_CLOUD_KEYFILE_JSON,
// env GCLOUD_KEYFILE_JSON,
// file ~/.gcp/osServiceAccount.json, and
// gcloud cli defaults

If none of the above are succesful in loading the credentials, it prompts the user to provide the location of the service account json or
paste the content to the prompt.

abhinavdahiya committed:
vendor: github.com/openshift/cluster-api-provider-gcp

abhinavdahiya committed:
asset/machines: add machine{,sets} for GCP

Similar to AWS, it creates

* a master machine in each Zone that's UP in the region
* a machineset for each Zone that's UP in the region and striping the required replicas over the set of machinesets.

abhinavdahiya committed:
types/gcp: Add set method to merge pools

abhinavdahiya committed:
pkg: create tfvars for gcp

* Uses the master machine objects to create the tfvars for terraform
* uses the credentials in the installer to pass through to terraform for authentication

jstuever committed:
google: create virtual machines

This change adds the terraform bits to create the bootstrap and control
plane virtual machines for the google platform. It also creates groups
for use by the load balancers with one for bootstrap and one for each
control plane zone.

abhinavdahiya committed:
data/data/rhcos.json: update the bootimage to 420.8.20190708.2 for CRI-O 1.14

updates the rhcos.json from https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.2/420.8.20190708.2/meta.json

abhinavdahiya committed:
asset/installlconfig/platformcredscheck.go: ensure the creds can atleast be loaded for GCP

abhinavdahiya committed:
data/data/gcp: make sure all gcp variables are prefixed with gcp

csrwng committed:
Add GCP uninstaller

Searches for most resources by name prefix using the cluster's Infra ID
For service accounts, it uses the displayName instead of the account
name, since that will contain the full Infra ID.
For load balancers dynamically created by the kube cloud controller, it
finds them indirectly through references to the Infra ID prefixed
instances.

openshift-merge-robot committed:
Merge pull request #1940 from jstuever/cors1119

google: create virtual machines

openshift-merge-robot committed:
Merge pull request #1875 from kikisdeliveryservice/remove-mco-component-images

bootkube.sh: remove image refs for mco components

jcpowermac committed:
Alternative: Add cloud providers metadata ip address to no proxy

- Checks the platform if it is not vSphere or None will
include 169.254.169.254 to the NO_PROXY list

openshift-merge-robot committed:
Merge pull request #1944 from abhinavdahiya/pr-1927

Create GCP specific assets: infrastructure, cred secrets, & dns

openshift-merge-robot committed:
Merge pull request #1945 from jcpowermac/alt_no_proxy_metadata

Add cloud providers metadata ip address to no proxy

tomassedovic committed:
data/data/openstack: Generate master clustervars

These variables are going to be used by the machine-config-operator to
seed the initial DNS and LB static pod configuration on each master.

stbenjam committed:
vendor: add terraform-provider-ironic for baremetal platform

This vendors terraform-provider-ironic and related code for use by the
baremetal IPI platform.

stbenjam committed:
data/baremetal: add terraform configuration for baremetal platform

This adds the terraform configuration for the baremetal platform. The
baremetal platform uses libvirt to create the bootstrap VM on the
deployment host, and then uses terraform-provider-ironic to provision
the baremetal machines for the control plane.

csrwng committed:
Vendor GCP IAM and Storage API

openshift-merge-robot committed:
Merge pull request #1937 from abhinavdahiya/gcp_machinessets

Add machine and machinesets for GCP

openshift-merge-robot committed:
Merge pull request #1563 from dcbw/ovn-ports

aws: allow GENEVE (6081) and OVN database ports (6641 & 6642)

openshift-merge-robot committed:
Merge pull request #1947 from tomassedovic/openstack-clustervars

data/data/openstack: Generate master clustervars

openshift-merge-robot committed:
Merge pull request #1938 from abhinavdahiya/gcp_tfvars

pkg: create tfvars for gcp

zeenix committed:
Make `terraform fmt` output diff

So that the output is more useful and actionable.

openshift-merge-robot committed:
Merge pull request #1957 from zeenix/tf-fmt

Make `terraform fmt` output diff

patrickdillon committed:
Add terminal prompts for GCP: region, project id, & base domain.

Add interactive terminal prompts to allow users to select region, project id and base domain when using the GCP platform. Automatically populate available regions and base domains based on project. Validate project ID.

openshift-merge-robot committed:
Merge pull request #1941 from abhinavdahiya/bump_rhcos_for_crio

data/data/rhcos.json: update the bootimage to 420.8.20190708.2 for CRI-O 1.14

openshift-merge-robot committed:
Merge pull request #1950 from csrwng/gcp_destroy

Uninstaller for GCP

openshift-merge-robot committed:
Merge pull request #1958 from openshift-metal3/vendor-baremetal

baremetal: vendor terraform-provider-ironic and provide initial terraform configuration

csrwng committed:
Fix gcp machine decoding

csrwng committed:
[gcp] pkg/asset/cluster - add gcp metadata

openshift-merge-robot committed:
Merge pull request #1964 from csrwng/fix_gcpmachine_type

[gcp] pkg/asset/machines: fix gcp machine decoding

openshift-merge-robot committed:
Merge pull request #1966 from csrwng/metadata_asset

[gcp] pkg/asset/cluster - add gcp metadata

openshift-merge-robot committed:
Merge pull request #1961 from patrickdillon/gcp-prompts-cors-1115

Add terminal prompts for GCP: region, project id, & base domain.

Commit:
Merge latest openshift/installer release

Commit:
Fix rebase mistake in data/data/baremetal/main.tf

hardys commented on data/data/baremetal/main.tf:
```diff
@@ -7,6 +7,11 @@ provider "ironic" {
+  url          = var.ironic_uri
+  microversion = "1.52"
+}
+
```
This is a rebase mistake

hardys commented:
@stbenjam and @markmc this has been tested via https://github.com/openshift-metal3/dev-scripts/pull/672 and appears to be working, if you get time to review that would be much aprpeciated :)

stbenjam commented on data/data/baremetal/main.tf:
```diff
@@ -20,6 +20,7 @@ module "bootstrap" {
 module "masters" {
   source = "./masters"
 
+  ironic_uri     = var.ironic_uri
```
I think this shouldn't be here, we removed it in our repo but not from openshift/installer yet

hardys replied:
Ah Ok, will fix and re-test

hardys replied:
Thanks, fixed

stbenjam commented on data/data/baremetal/masters/variables.tf:
```diff
@@ -1,8 +1,14 @@
+  type        = string
+  description = "URI for Ironic that bootstrap should wait for to be available"
+}
+
```
Same here

hardys replied:
Thanks, fixed

stbenjam reviewed (commented):
Otherwise I think this looks good

Commit:
Fix rebase mistake re-adding ironic_uri

This was removed by #137 and mistakenly re-added during
rebase to upstream.

stbenjam commented on pkg/terraform/exec/plugins/Gopkg.lock:
```diff
@@ -795,6 +795,14 @@
 
+[[projects]]
+  digest = "1:a0886267c60262a245d6eeb0be12bf8e165026ad088bce106b9c55c5d613144f"
+  name = "github.com/openshift-metalkube/terraform-provider-ironic"
```
Ooh sorry looks like this got included too, we should only use the metal3 version, I'd remove this from Gopkg.lock and the openshift-metalkube directory in vendor

Commit:
Remove unused openshift-metalkube/terraform-provider-ironic

This is another rebase mistake, we should instead use the upstream
openshift-metal3/terraform-provider-ironic copy.

stbenjam reviewed (approved):
lgtm

stbenjam commented:
CI passed going to merge this.
